test/e2e/handler/nns_update_timestamp_test.go

```diff
@@ -17,30 +18,43 @@ import (
 var _ = Describe("[nns] NNS LastSuccessfulUpdateTime", func() {
 	var (
 		originalNNSs map[string]nmstatev1beta1.NodeNetworkState
-	)
-	BeforeEach(func() {
-		originalNNSs = map[string]nmstatev1beta1.NodeNetworkState{}
-		for _, node := range allNodes {
-			key := types.NamespacedName{Name: node}
-			originalNNSs[node] = nodeNetworkState(key)
+		nnsByNode    = func() map[string]nmstatev1beta1.NodeNetworkState {
+			NNSs := map[string]nmstatev1beta1.NodeNetworkState{}
+			for _, node := range allNodes {
+				key := types.NamespacedName{Name: node}
+				NNSs[node] = nodeNetworkState(key)
+			}
+			return NNSs
 		}
-	})
+	)
 	Context("when network configuration hasn't change", func() {
 		It("should not be updated", func() {
-			By("Give enough time for the NNS Reconcile to happend (3 interval times)")
-			time.Sleep(3 * nmstatenode.NetworkStateRefresh)
-			for node, originalNNS := range originalNNSs {
-				obtainedNNSStatus := nodeNetworkState(types.NamespacedName{Name: node}).Status
-				obtainedState := state.RemoveDynamicAttributesFromStruct(obtainedNNSStatus.CurrentState)
-				originalState := state.RemoveDynamicAttributesFromStruct(originalNNS.Status.CurrentState)
-				Expect(obtainedState).To(Equal(originalState), "should report same state, diff :%s", diff.LineDiff(originalState, obtainedState))
-				Expect(obtainedNNSStatus.LastSuccessfulUpdateTime).To(Equal(originalNNS.Status.LastSuccessfulUpdateTime))
-				Expect(obtainedNNSStatus.Conditions).To(Equal(originalNNS.Status.Conditions))
-			}
+			Eventually(func() error {
```

Comment:
if you say that the `currentState` gets changed by the system so NNS status gets refreshed, then I'm perplexed - isn't this what the test is specifically checking? 
Can you say what get's refreshed by the system exactly, so you could distinguish between "innocent refreshes made by the system" and the ones this test should fail upon?

if not, then I don't see how adding eventually will help us.

Reply:
Sometimes a new route is added by the system, I suspect is par of teardown code that reset nics, so we retry, if one of the retry passes is good since it means that multiple calls to Reconcile didn't update the NNSs status if not needed.

Reply:
if the currentState is change by the system, then from then on all the retries will fail, or is it more comlpex than that?

Reply:
After running it several times sometimes it needs just another retry and then is fine, looks like it's unlikely it happens twice in a row.

Reply:
ok talking offline I understand that the system updates seem to be added only in a transitional state.
In that case then I agree that reiterating the state with Eventually until we see that the state is stable is good.